### PR TITLE
Feature/end points mapoteca

### PR DESCRIPTION
**Pull request by smartsena**

### CODE REVIEW

File: documental/models.py [hunk 1]
```diff
@@ -183,15 +197,113 @@ class DocumentalDocs(models.Model):
...
+    )
+
+    map_dimension = models.CharField(
+        _('Map page dimension'),
+        max_length=2,
+        blank=True,
+        null=True,
+    )
+
+    # from django.contrib.postgres.fields.jsonb import JSONField
```
Comment (guigawiller) [inline documental/models.py:260]: Remove unnecessary comments.

File: documental/models.py [hunk 2]
```diff
@@ -183,15 +197,113 @@ class DocumentalDocs(models.Model):
...
+        verbose_name_plural = 'Documents Indigenous Lands'
+
+    def __str__(self) -> str:
+        """Returns `documental.DocsDocumentTI` string data.
         Returns:
-            str: model data path_documento.
+            str: model data document name.
         """
-        return self.path_document
+        return self.no_document
```
Comment (guigawiller) [inline documental/models.py:240]: Missing docstrings end.

File: documental/models.py [hunk 3]
```diff
@@ -183,15 +197,113 @@ class DocumentalDocs(models.Model):
...
+        """Returns `documental.DocsMapoteca` string data.
+        Returns:
+            str: model data document name.
+        """
+        return self.no_document
+
+
+class DocumentUpload(models.Model):
+
+    # uploaded_at = models.DateTimeField(auto_now_add=True)
```
Comment (guigawiller) [inline documental/models.py:288]: Removes unnecessary coments.

File: documental/models.py [hunk 4]
```diff
@@ -183,15 +197,113 @@ class DocumentalDocs(models.Model):
...
+    
+    file = models.FileField(upload_to='specs')
+
```
Comment (guigawiller) [inline documental/models.py:291]: Trailing whitespaces.

File: documental/models.py [hunk 5]
```diff
@@ -183,15 +197,113 @@ class DocumentalDocs(models.Model):
...
+
+class DocumentUpload(models.Model):
+
+    # uploaded_at = models.DateTimeField(auto_now_add=True)
+    
+    file = models.FileField(upload_to='specs')
+    
+    dt_cadastro = models.DateTimeField( #dt_cadastro
+        _('Document registration date'),
```
Comment (guigawiller) [inline documental/models.py:293]: Trailing whitespaces.

File: documental/models.py [hunk 6]
```diff
@@ -183,15 +197,113 @@ class DocumentalDocs(models.Model):
...
+    )
+    
+    id_acao = models.ForeignKey(
+        'documental.DocsAction',
+        on_delete=models.DO_NOTHING,
+        related_name='documentosdocs_action',
+        null=True
+    )
+    class Meta:
+        # ordering = ['uploaded_at']
```
Comment (guigawiller) [inline documental/models.py:305]: removes unnecessary coments.

File: documental/serializers.py [hunk 7]
```diff
@@ -32,42 +33,42 @@ class Meta:
...
             'st_excluded',
+            'dt_registration',
+            'dt_update',
             'co_funai',
-            'no_ti',
             'co_cr',
             'ds_cr',
-            'dt_registration',
-            'dt_update',
+            # 'no_ti',
```
Comment (guigawiller) [inline documental/serializers.py:59]: Removes unnecessary coments.

File: documental/serializers.py [hunk 8]
```diff
@@ -76,42 +77,88 @@ def to_representation(self, instance):
...
-            'no_extension',
             'st_available',
             'st_excluded',
+            'dt_registration',
+            'dt_update',
             'co_funai',
-            'no_ti',
             'co_cr',
             'ds_cr',
+            # 'no_ti',
```
Comment (guigawiller) [inline documental/serializers.py:103]: Removes unecessary coments.

File: documental/serializers.py [hunk 9]
```diff
@@ -76,42 +77,88 @@ def to_representation(self, instance):
...
+            'no_document',
+            'st_available',
+            'st_excluded',
             'dt_registration',
             'dt_update',
-            'dt_document',
+            'co_funai',
+            'co_cr',
+            'ds_cr',
+            # 'no_ti',
```
Comment (guigawiller) [inline documental/serializers.py:148]: Removes unnecessary coments.

File: documental/serializers.py [hunk 10]
```diff
@@ -76,42 +77,88 @@ def to_representation(self, instance):
...
+
+    def to_representation(self, instance):
+        """Method to return in `DocsDocumentTISerializers` the full URL to
+        download the documents in `models.DocsDocumentTI`.
+
+        Returns:
+            str: url to DocumentTI document
+        """
+
+        url_document = super().to_representation(instance)
```
Comment (guigawiller) [inline documental/serializers.py:119]: Check if this is really necessary.

File: documental/urls.py [hunk 11]
```diff
@@ -12,5 +12,10 @@
         'list/',
         views.DocumentalListViews.as_view(),
         name='list-doc'
+    ),
```
Comment (guigawiller) [inline documental/urls.py:15]: Check if this is really necessary.

File: documental/views.py [hunk 12]
```diff
@@ -22,46 +22,110 @@ class AuthModelMix:
...
+        elif action_type_docs == "MAPS_LAND_USER":
+            self.filterset_class = documental_filters.DocsLandUserFilter
+            return models.DocsLandUser.objects.all()
+        elif action_type_docs == "MAPOTECA":
+            return models.DocsMapoteca.objects.all()
         else:
             raise exceptions.ParseError(
-                "Não permitido retorno de dados de DocumentoTI"
-                " UsoEOcupaçãoDoSolo na mesma requisição", None)
+                "ERRO in the data set returned", None)
```
Comment (guigawiller) [inline documental/views.py:131]: Change this to ERROR.

File: documental/filters.py [hunk 13]
```diff
@@ -19,16 +19,26 @@ class CharInFilter(
     pass
 
 
+class DocsActionFilter(rest_framework.FilterSet):
+    action_type = rest_framework.CharFilter(
```
Comment (guigawiller) [inline documental/filters.py:23]: Missing docstrings.

File: documental/filters.py [hunk 14]
```diff
@@ -19,16 +19,26 @@ class CharInFilter(
...
+        field_name='action_type',
+        # required=True,
+    )
+
+    class Meta:
```
Comment (guigawiller) [inline documental/filters.py:28]: Missing docstrings.

File: documental/admin.py [hunk 15]
```diff
@@ -3,13 +3,32 @@
 from documental import models
 
 
+list_documental_abstract_class = [
```
Comment (guigawiller) [inline documental/admin.py:6]: Move this to project settings.

File: documental/admin.py [hunk 16]
```diff
@@ -31,34 +50,63 @@ class UsuarioAdmin(admin.ModelAdmin):
...
+    ] + list_documental_abstract_class
+
+    fields = list_display
+
+    search_fields = list_display
+
+
+class DocumentUploadAdmin(admin.ModelAdmin):
+    list_display = [
+        # 'uploaded_at',
```
Comment (guigawiller) [inline documental/admin.py:95]: Removes unnecessary coments.

File: documental/models.py (hunk 1 above)
Comment (smartsena) [reply documental/models.py:260]: Removed.

File: documental/models.py (hunk 2 above)
Comment (smartsena) [reply documental/models.py:240]: Adjusted.

File: documental/models.py (hunk 3 above)
Comment (smartsena) [reply documental/models.py]: Removed.

File: documental/models.py (hunk 4 above)
Comment (smartsena) [reply documental/models.py]: Removed.

File: documental/models.py (hunk 5 above)
Comment (smartsena) [reply documental/models.py]: Removed code class.

File: documental/models.py [hunk 17]
```diff
@@ -183,15 +197,113 @@ class DocumentalDocs(models.Model):
...
+    file = models.FileField(upload_to='specs')
+    
+    dt_cadastro = models.DateTimeField( #dt_cadastro
+        _('Document registration date'),
+        null=True,
+        blank=True,
+    )
+    
+    id_acao = models.ForeignKey(
+        'documental.DocsAction',
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/models.py]: Removed.

File: documental/models.py (hunk 6 above)
Comment (smartsena) [reply documental/models.py]: Removed.

File: documental/serializers.py (hunk 7 above)
Comment (smartsena) [reply documental/serializers.py:59]: Uncommented code.

File: documental/serializers.py (hunk 8 above)
Comment (smartsena) [reply documental/serializers.py:103]: Uncommented code.

File: documental/serializers.py (hunk 9 above)
Comment (smartsena) [reply documental/serializers.py:148]: Uncommented code.

File: documental/serializers.py (hunk 10 above)
Comment (smartsena) [reply documental/serializers.py:119]: It's necessary. Follow documentation link used to create this code:

https://www.django-rest-framework.org/api-guide/serializers/#advanced-serializer-usage

File: documental/views.py (hunk 12 above)
Comment (smartsena) [reply documental/views.py:131]: Adjusted.

File: documental/filters.py (hunk 13 above)
Comment (smartsena) [reply documental/filters.py:23]: Created docstrings.

File: documental/admin.py (hunk 15 above)
Comment (smartsena) [reply documental/admin.py]: Will be organized in another demand.

File: documental/admin.py (hunk 16 above)
Comment (smartsena) [reply documental/admin.py]: Removed.

File: documental/models.py [hunk 18]
```diff
@@ -64,6 +72,7 @@ class Meta:
         app_label = 'documental'
         verbose_name = 'User CMR'
         verbose_name_plural = 'Users CMR'
+        # usuario = models.ForeignKey(settings.AUTH_USER_MODEL, related_name=_('usuario'))
```
Comment (guigawiller) [inline documental/models.py:75]: Remove unnecessary comments

**Review by guigawiller — CHANGES_REQUESTED** [review]
Remove the file documental/app_settings.py

File: documental/models.py (hunk 18 above)
Comment (smartsena) [reply documental/models.py:75]: Removed.

File: documental/filters.py (hunk 14 above)
Comment (smartsena) [reply documental/filters.py:33]: Removed.

File: documental/admin.py [hunk 19]
```diff
@@ -31,34 +50,63 @@ class UsuarioAdmin(admin.ModelAdmin):
...
+        'map_dimension',
+        'js_ti',
+    ] + list_documental_abstract_class
+
+    fields = list_display
+
+    search_fields = list_display
+
+
+class DocumentUploadAdmin(admin.ModelAdmin):
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/admin.py]: Removed.

**Review by guigawiller — APPROVED** [review]

**Pull request merged**